schemaregistry/serde/serde.go

```diff
@@ -0,0 +1,218 @@
+package serde
+
+import "C"
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"github.com/confluentinc/confluent-kafka-go/schemaregistry"
+)
+
+// Type represents the type of Serde
+type Type = int
+
+const (
+	// KeySerde denotes a key Serde
+	KeySerde = 1
+	// ValueSerde denotes a value Serde
+	ValueSerde = 2
+)
+
+const (
+	// EnableValidation enables validation
+	EnableValidation = true
+	// DisableValidation disables validation
+	DisableValidation = false
+)
+
+// MagicByte is prepended to the serialized payload
+const MagicByte byte = 0x0
```

Comment:
probably doesn't need to be public, i think probably make it private. if public, name could be something like `ConfluentFramingMagicByte`, or something better, don't know what.